h5c/vic/src/vic-webapp/src/app/container-view/container-view.template.html

```diff
@@ -1,4 +1,30 @@
-<clr-datagrid (clrDgRefresh)="refreshGrid($event)" [clrDgLoading]="isDgLoading">
+<div class="col-xs-12 mb-2" *ngIf="errorObj">
+    <div class="row">
+      <div class="col-xs-12">
+        <clr-alert [clrAlertType]="'danger'"
+                  [clrAlertClosable]="false">
+          <clr-alert-item *ngIf="errorObj.type === 'ssl_cert'">
+            <span class="alert-text">
+              <!-- TODO: i18n -->
+              Failed to verify the VIC machine server at endpoint https://{{ errorObj.payload }}:8443. Your browser may be blocking the connection because of the untrusted SSL certificate.
```

Comment:
It seems like we do not have enough information to distinguish between:
 * Unavailability of the API (e.g., it crashed).
 * Networking configuration issues (e.g., the OVA is on a management network that isn't generally accessible).
 * SSL configuration issues.

I think SSL is most likely, but I think it's worth clearly explaining it's just one of several potential problems. We don't want people to waste a lot of time trying to debug SSL if it's not the problem.

Reply:
If API is unavailable or unreachable or something else we will get a proper HTTP response status code like 4xx or 5xx, but in the case of this we get status code of 0. Here's the logic that returns `errObj.type = 'ssl_cert'`

https://github.com/vmware/vic-ui/pull/106/files#diff-f361ad37b31a58cab387befb69ffe05aR363

Let me know your thought.

Reply:
Have you actually tested with an unreachable or crashed API sever? You won't get a 4xx or 5xx in either case, since the server isn't reachable.

Reply:
Angular's HttpClient handles it well, and as you can see in the example below a call to the endpoint that is unreachable returns 0 status code.

![image](https://user-images.githubusercontent.com/3834071/33145577-8985ac6e-cf86-11e7-848c-6f4fa6fe9656.png)

Reply:
This implies that there's a caveat to this approach that 0 can mean not just unsafe SSL cert but also unreachable host. I forgot to update the error message before merging. But will get that in as a separate commit to another PR.